lms/djangoapps/courseware/tests/test_video_mongo.py

```diff
@@ -1107,7 +1141,7 @@ def test_with_edx_video_id(self, allow_cache_miss, video_exists_in_val, associat
         if video_exists_in_val and (associate_course_in_val or allow_cache_miss):
             self.verify_result_with_val_profile(result)
         else:
-            self.verify_result_with_fallback_url(result)
+            self.verify_result_with_fallback_and_youtube(result)
```

Comment:
This test has too many branches, and an opaque DDT data section.  Can you pull it apart into separate tests that don't have conditionals all over the place?